lib/aurora_bootstrapper/notifier.rb

```diff
@@ -12,7 +12,48 @@ def export_date
     
     # ENV is string to string dictionary
     def export_date_override
-      DateTime.now.strftime("%Y-%m-%d") if ENV.key?('EXPORT_DATE_OVERRIDE')
+      date_override = nil
+
+      if ENV.key?('EXPORT_DATE_OVERRIDE')
+        now = Date.today
+
+        # expiration time is 30 days
+        for i in 1..30
+          if exists_db_dump_done_for_one_day?( date: (now-i))
+            date_override = convert_date_to_string(date: (now-i))
+            break
+          end
+        end
+
+        # the first run for the given db partition
+        if date_override.nil?
+          date_override = convert_date_to_string(date: now)
+        end
+      end
+
+      date_override
+    end
+
+    def exists_db_dump_done_for_one_day?( date: )
```

Comment:
can we call this something like `exists_export?( date: )`

Reply:
yes